### DISCUSSION TITLE
Make more use of STEP_ID_INIT and STEP_ID_USER constants

### DISCUSSION THREAD

**Discussion by epenet** [home-assistant/architecture#660] [General] [answered]
Pull requests https://github.com/home-assistant/core/pull/58590 and https://github.com/home-assistant/core/pull/58602 introduced two new constants in `homeassistant/data_entry_flow.py`:
```python
STEP_ID_INIT: Final = "init"
STEP_ID_USER: Final = "user"
```

Most components use this just once or twice in `config_flow.py`, but then use them loads of times in the `test_config_flow.py`

I propose that all the `config_flow.py` and `test_config_flow.py` code using "init" and "user" strings for the step ID is updated to use the constants.

Draft PRs:
- https://github.com/home-assistant/core/pull/58681
- https://github.com/home-assistant/core/pull/58692
- https://github.com/home-assistant/core/pull/58693
- https://github.com/home-assistant/core/pull/58694
- https://github.com/home-assistant/core/pull/58695
- https://github.com/home-assistant/core/pull/58696

**Comment by MartinHjelmare** [comment, marked as answer]
There are a couple of reasons why I think we should not add constants for the step ids.

The steps in the config flow are defined by each integration. It is true that the start steps (user/init) are decided by the config entries / data entries API. But all the other steps are decided by the integration. Having constants for some steps and not for other steps is weird and inconsistent. I don't think we should add constants in each integration for all step ids either since the step ids are only used in the config flow module and should preferably only be used once or twice. If a step id is used more than that, it's a sign that the flow probably needs to be refactored.

A constant for the step id also doesn't help the developer very much since care must still be taken to use the constant in the correct step. Each step function must only show the form of its step method. The developer can not simply use the constant from a common function and get the correct form.

I don't think we should add constants in the production code just to be able to use them as assert targets in the tests.
